contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -305,6 +350,53 @@ contract MixinSignatureValidator is
         ));
     }
 
+    /// @dev Verifies signature using logic defined by an EIP1271 Wallet contract.
+    /// @param hash Any 32 byte hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own signature verification method.
+    /// @param signature Proof that the hash has been signed by signer.
+    /// @return True if the signature is validated by the Walidator.
+    function _validateHashWithEIP1271Wallet(
+        bytes32 hash,
+        address walletAddress,
+        bytes memory signature
+    )
+        private
+        view
+        returns (bool isValid)
+    {
+        uint256 signatureLength = signature.length;
+        // Shave the signature type off the signature.
+        assembly {
+            mstore(signature, sub(signatureLength, 1))
+        }
+        // Encode the call data.
+        bytes memory data = new bytes(32);
+        data.writeBytes32(0, hash);
+        bytes memory callData = abi.encodeWithSelector(
+            IEIP1271Wallet(walletAddress).isValidSignature.selector,
```

Comment:
Can we hard code these selectors somewhere? (or at least check what the gas differences are).

Reply:
If Remix is to be believed, on `0.5.10`, this way is weirdly cheaper *both* with and without optimizations by 28 and 8 gas, respectively.  :woman_shrugging:

Reply:
Bizarre... going to add a task to test this throughout the codebase. It's nice not having to maintain all of these contracts that just contain constants.